api/src/main/java/org/openmrs/module/bedmanagement/service/impl/BedManagementServiceImpl.java

```diff
@@ -224,7 +229,10 @@ public Bed getBedByUuid(String uuid) {
 	}
 	
 	@Override
-	public void deleteBed(Bed bed, String reason) {
+	public void deleteBed(Bed bed, String reason) throws BedOccupiedException {
+		if (BedStatus.OCCUPIED.toString().equals(bed.getStatus())) {
```

Comment:
Ok that's simpler than what I thought. Obviously we rely on the fact that `bed.getStatus()` is always accurate.
It _should_ be, but I'm actually worried that there might be other loopholes allowing that flag to be out of sync.

Wasn't it safer to look at `BedPatientAssignment` to be 100% sure that a bed is assigned or not to a patient?

Reply:
I also hadn't realized it would be this easy.
This status field that is updated via AOP, so it seems good enough to me. (Of course if someone edits the DB outside of the API this would be a problem, but I don't think we need to worry about that.)